kasa/smart/smartchilddevice.py

```diff
@@ -107,16 +107,26 @@ async def create(
     @property
     def device_type(self) -> DeviceType:
         """Return child device type."""
-        category = self.sys_info["category"]
-        dev_type = self.CHILD_DEVICE_TYPE_MAP.get(category)
-        if dev_type is None:
-            _LOGGER.warning(
-                "Unknown child device type %s for model %s, please open issue",
-                category,
-                self.model,
-            )
-            dev_type = DeviceType.Unknown
-        return dev_type
+        if self._device_type is not DeviceType.Unknown:
+            return self._device_type
+
+        if self.sys_info and (category := self.sys_info.get("category")):
+            dev_type = self.CHILD_DEVICE_TYPE_MAP.get(category)
+            if dev_type is None:
```

Comment:
```suggestion
            self._device_type = self.CHILD_DEVICE_TYPE_MAP.get(category)
            if self._device_type is None:
```
Setting the private attribute directly and handling the none case separately allows simplifying this piece of code.

Reply:
Reverted this change as it breaks mypy because `self.device_type` is not nullable and I don't think we want to change that.

Reply:
Agreed, we should avoid nullables where possible :+1: